src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/StreamingCounterLogger.cs

```diff
@@ -3,39 +3,71 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.Diagnostics.Monitoring.EventPipe;
-using System.IO;
+using System;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
 
 namespace Microsoft.Diagnostics.Monitoring.WebApi
 {
     internal abstract class StreamingCounterLogger : ICountersLogger
     {
-        private readonly Stream _stream;
+        private const int BoundedChannelSize = 1000;
```

Comment:
Can you test if we drop anything in a resource constrained environment such as k8 with our resource recommendations + 1 sec metric interval? I'm expecting that we don't but it might help guide the optimal size of this buffer.

Reply:
I'd like to defer this until after the changes are merged at this point, but I will definitely check this, probably add some more diagnostic logging to see how full the channel gets with different providers and numbers of counters.